ikarus/finiteelements/mechanics/loads/traction.hh

```diff
@@ -0,0 +1,192 @@
+         * @param tractionBoundary The element boundary where traction load is applied.
+         * @param force The force vector to be updated.
+         * @param dx Optional displacement vector.
+         */
+        template<typename ScalarType, typename BoundaryType>
+        void calculateVector(const FERequirementType &par, const BoundaryType &tractionBoundary,
+                             typename Traits::template VectorType<ScalarType> force,
+                             const std::optional<const Eigen::VectorX<ScalarType>> dx = std::nullopt) {
+            using namespace Dune::DerivativeDirections;
+            using namespace Dune;
+            const auto uFunction = displacementFunction(par, dx);
+            const auto &lambda = par.getParameter(Ikarus::FEParameter::loadfactor);
+            auto element = this->localView().element();
+            for (auto &&intersection: intersections(tractionBoundary->gridView(), element)) {
+                if (not tractionBoundary->contains(intersection)) continue;
+
+                /// Integration rule along the boundary
+                const auto &quadLine = Dune::QuadratureRules<double, myDim - 1>::rule(intersection.type(), order);
+
+                for (const auto &curQuad: quadLine) {
+                    const Dune::FieldVector<double, myDim> &quadPos = intersection.geometryInInside().global(
+                            curQuad.position());
+                    const double intElement = intersection.geometry().integrationElement(curQuad.position());
+
+                    /// The value of the local function wrt the i-th coeff
+                    for (size_t i = 0; i < numberOfNodes; ++i) {
+                        const auto udCi = uFunction.evaluateDerivative(quadPos, wrt(coeff(i)));
+
+                        /// Value of the Neumann data at the current position
+                        auto neumannValue = neumannBoundaryLoad(intersection.geometry().global(curQuad.position()),
+                                                                lambda);
+                        force.template segment<worldDim>(worldDim * i) -=
+                                udCi * neumannValue * curQuad.weight() * intElement;
+                    }
+                }
+            }
+        }
+
+        /**
+           * @brief Calculates the matrix stiffness for the given FERequirementType.
+           *
+           * @param par The FERequirementType object.
+           * @param K Matrix to store the calculated stiffness.
+           */
+        template<typename ScalarType>
+        void calculateMatrix(const FERequirementType &par, typename Traits::template MatrixType<> K,
+                             const std::optional<const Eigen::VectorX<ScalarType>> &dx = std::nullopt) const {
+            DUNE_THROW(Dune::NotImplemented, "calculateMatrix is not implemented for the class Volume");
+        }
+
+        bool isApplied() { return isApplied_; }
```

Comment:
This should not be necessary. If trhe functions do not have anything todo do an early exit here.